packages/pds/src/read-after-write/util.ts

```diff
@@ -45,9 +45,23 @@ export const readAfterWriteInternal = async <T>(
   res: HandlerPipeThrough,
   munge: MungeFn<T>,
 ): Promise<HandlerResponse<T> | HandlerPipeThrough> => {
-  const rev = getRepoRev(res.headers ?? {})
+  const rev = getRepoRev(res.headers)
   if (!rev) return res
+
+  // If the response's "atproto-repo-rev" header matches the current repo rev,
+  // we can skip the munge step and return the response as-is.
+  const repoRev = await ctx.repoRevCache?.get(requester)
```

Comment:
similarly, it'd be great to fail open here, maybe with a quick timeout on the redis request